File: src/public/training/auth-training/kotlin-demo/demo-auth-app/src/main/kotlin/com/calian/at/demoAuth/solutions/lesson3_rbac/UserDatabase.kt
package com.calian.at.demoAuth.solutions.lesson3_rbac

interface UserDatabaseInterface {
    fun getUser(username: String): User?
    fun addUser(user: User)
    fun verifyUser(username: String, password: String): Boolean
}

class UserDatabase : UserDatabaseInterface {

    // Pretend we are a database.
    private val users = mutableListOf<User>(
        User("Quinn", "Bast", listOf(Role.ADMIN, Role.USER)),
        User("Alice", "Wonderland", listOf(Role.VIEWER)),
        User("Bobby", "Tables", listOf(Role.ADMIN)),
    )

    override fun getUser(username: String): User? {
        return users.find { it.username == username }
    }

    override fun addUser(user: User) {
        users.add(user)
    }

    override fun verifyUser(username: String, password: String): Boolean {
        return users.any { it.username == username && it.password == password }
    }
}

File: src/public/training/auth-training/csharp-demo/AuthTraining/Controllers/lesson2/UserDatabase.cs
namespace AuthTraining.Controllers.lesson2;

public interface IUserDatabase
{
    public void AddUser(User user);
    public User GetUser(string username);
    public bool VerifyUser(string username, string password);
}

public class UserDatabase : IUserDatabase
{

    private List<User> Users { get; } =
    [
        new User { username = "Quinn", password = "Bast" },
        new User { username = "Alice", password = "Wonderland" },
        new User { username = "Bobby", password = "Tables" },
    ];
    
    public void AddUser(User user)
    {
        Users.Add(user);
    }

    public User GetUser(string username)
    {
        return Users.Find((user) => user.username == username);
    }

    public bool VerifyUser(string username, string password)
    {
        return Users.Any((user) => user.username == username && user.password == password);
    }
}